sphinx/pycode/__init__.py

```diff
@@ -124,7 +124,7 @@ def for_module(cls, modname: str) -> "ModuleAnalyzer":
         cls.cache['module', modname] = obj
         return obj
 
-    def __init__(self, source: IO, modname: str, srcname: str) -> None:
+    def __init__(self, source: IO[Any], modname: str, srcname: str) -> None:
```

Comment:
I think this is the better API, and then put the `.read()` calls in the class methods.

```suggestion
    def __init__(self, source: str, modname: str, srcname: str) -> None:
```

Reply:
what would that look like inside the method?

Reply:
`self.code = source.read()` becomes `self.code = source`

A

Reply:
oh i see, i thought you meant do the `read` in the `__init__` method. Yep, i agree with you. Will update.